text/0000-rust-governance.md

```diff
@@ -0,0 +1,728 @@
+which should be prioritized.
+
+### Moderation team
+
+Finally, the moderation team is responsible for dealing with CoC violations.
+
+One key difference from the other subteams is that the moderation team does not
+have a leader. Its members are chosen directly by the core team, and should be
+community members who have demonstrated the highest standard of discourse and
+maturity. To limit conflicts of interest, **the moderation subteam should not
+include any core team members**. However, the subteam is free to consult with
+the core team as it deems appropriate.
+
+The moderation team will have a public email address that can be used to raise
+complaints about CoC violations (forwards to all active moderators).
+
+#### Initial plan for moderation
+
+What follows is an initial proposal for the mechanics of moderation. The
+moderation subteam may choose to revise this proposal by drafting an RFC, which
+will be approved by the core team.
+
+Moderation begins whenever a moderator becomes aware of a CoC problem, either
+through a complaint or by observing it directly. In general, the enforcement
+steps are as follows:
+
+> **These steps are adapted from text written by Manish Goregaokar, who helped
+articulate them from experience as a Stack Exchange moderator.**
+
+* Except for extreme cases (see below), try first to address the problem with a
+  light public comment on thread, aimed to de-escalate the situation. These
+  comments should strive for as much empathy as possible. Moderators should
+  emphasize that dissenting opinions are valued, and strive to ensure that the
+  technical points are heard even as they work to cool things down.
+
+  When a discussion has just gotten a bit heated, the comment can just be a
+  reminder to be respectful and that there is rarely a clear "right" answer. In
+  cases that are more clearly over the line into personal attacks, it can
+  directly call out a problematic comment.
+
+* If the problem persists on thread, or if a particular person repeatedly comes
+  close to or steps over the line of a CoC violation, moderators then email the
+  offender privately. The message should include relevant portions of the CoC
+  together with the offending comments. Again, the goal is to de-escalate, and
+  the email should be written in a dispassionate and empathetic way. However,
+  the message should also make clear that continued violations may result in a
+  ban.
+
+* If problems still persist, the moderators can ban the offender. Banning should
+  occur for progressively longer periods, for example starting at 1 day, then 1
+  week, then permanent. The moderation subteam will determine the precise
+  guidelines here.
+
+In general, moderators can and should unilaterally take the first step, but
+steps beyond that (particularly banning) should be done via consensus with the
+other moderators. Permanent bans require core team approval.
```

Comment:
> - the most important of which, historically, is not the claim that the core team is breaking the CoC but rather that any attempts to enforce it are actually about squelching dissent on technical topics.

FWIW this is the situation I was considering when I noted that we should avoid Mozilla employees (/core team members) in the original CoC enforcement proposal. I did not think we would need to plan for a CT member violating the CoC, nor would we have to plan for the community being wary of the same. I still don't think we need to. I'm more wary of a perception of the core team squelching dissent, as you said.


> A ban might be reversible, but the fallout from making a ban, and then reversing it, could be extreme. Much better to avoid the mistake in the first place.

Huge +1 here. Bans can easily antagonize the recipient (that's why we try to resolve it without a ban first), make people feel patronized, and in general sour a situation further.

However, note that in most cases with bans time is of the essence. Given that the core team is concentrated on one hemisphere, waiting for CT approval on something might take a day -- bear in mind that it probably already took half a day for the mod team to discuss this out. This is a lot of time to wait if there's a fight brewing or something. I'm not so concerned about a perception of corruption due to this as I am concerned of the time lost in the process.

So it might be okay to be looser on the core team approval thing for a certain degree of severity.

Though the time thing becomes a non-issue if, after the first or second infraction by someone, the mod team and perhaps a core team member discuss it out internally and decide what to do the next time an infraction of the same type comes from that person, if ever. If there doesn't seem to be any disagreement, if the person violates the CoC again, a mod can just unilaterally-ish hand out a ban the next time.  (In cases where the pre-discussion isn't unanimous, it might be best to discuss it out)